src/utils/CCTPUtils.ts

```
@@ -237,10 +237,15 @@ async function _resolveCCTPRelatedTxns(
           if (processed) {
             status = "finalized";
           } else {
-            // Generate the attestation proof for the message. This is required to finalize the message.
-            attestation = await generateCCTPAttestationProof(messageHash, utils.chainIsProd(destinationChainId));
-            // If the attestation proof is pending, we can't finalize the message yet.
-            status = attestation.status === "pending_confirmations" ? "pending" : "ready";
+            // Try to generate the attestation proof for the message. This is required to finalize the message.
+            // If the request fails, which may happen if Circle's attestation API is falling behind, ignore the finalization on this run.
+            try {
+              attestation = await generateCCTPAttestationProof(messageHash, utils.chainIsProd(destinationChainId));
+              // If the attestation proof is pending, we can't finalize the message yet.
+              status = attestation.status === "pending_confirmations" ? "pending" : "ready";
+            } catch (e) {
+              return undefined;
```

I think we'd want to at least log something here because otherwise the failure is silently suppressed, so we might have a long period of issues trying to finalize before we actually notice it.

I note that we don't have a logger instance here and it's probably annoying to supply one, so I'm wondering if we should just catch this at the top level where we anyway need to catch and report on errors. wdyt?

---

If we throw an error here, then wouldn't this interrupt `resolveCCTPRelatedTxns` and basically cause all CCTP finalizations for the chain to be ignored? So for example, if we have 3 ready to finalize transactions, one of which returns a 404 from an API request, then wouldn't propagating that error up a level effectively cut off this function from returning the other two ready to finalize transactions? 

To be fair, I think this is possible with some refactoring, so then the question is, if we want to do this approach, should we just close this PR and make a new one which does this type of refactoring.

---

I added a top level try/catch here https://github.com/across-protocol/relayer/pull/1928/commits/8ac78ddf3b29b084d289ae8b8fa524f97cf9ac91. This should mean that if the finalizers fail sometime in the future for whatever reason, we can catch it here and still continue. I didn't get rid of the try/catch in CCTP since I still think the point stands that if we just throw here, then we will not finalize any more cross-chain transfers. I.e. if we have 5 ready-to-finalize transfers, but one of them errors on the axios request, then all five would be ignored if we just threw an error.

---

On second thought, it is best to fail loudly than fail silently. So this was removed https://github.com/across-protocol/relayer/pull/1928/commits/7b40c5173ebd2dec2f34b1011b4bd1b15d2a7b8e

---

We should avoid silently throwing errors. A simple debug log would help us detect failures over time. Failing silently risks missing CCTP finalization if the endpoint goes down without any GCP debug logs.